mono/metadata/exception.c

```diff
@@ -45,13 +50,78 @@ static gpointer unhandled_exception_hook_data = NULL;
  *
  * \returns the initialized exception instance.
  */
+static MonoExceptionHandle
+mono_exception_new_by_name (MonoImage *image, const char *name_space, const char *name, MonoError *error)
+{
+	return mono_exception_new_by_name_domain (mono_domain_get (), image, name_space, name, error);
+}
+
 MonoException *
 mono_exception_from_name (MonoImage *image, const char *name_space,
 			  const char *name)
 {
 	return mono_exception_from_name_domain (mono_domain_get (), image, name_space, name);
 }
 
+/**
+ * mono_exception_from_name_domain:
+ * \param domain Domain where the return object will be created.
+ * \param image the Mono image where to look for the class
+ * \param name_space the namespace for the class
+ * \param name class name
+ *
+ * Creates an exception object of the given namespace/name class on
+ * the given domain.
+ *
+ * \returns the initialized exception instance.
+ */
+static MonoExceptionHandle
+mono_exception_new_by_name_domain (MonoDomain *domain, MonoImage *image,
+				 const char* name_space, const char *name, MonoError *error)
+{
+	HANDLE_FUNCTION_ENTER ()
+
+	MonoClass *klass;
+	MonoObjectHandle o = MONO_HANDLE_NEW (MonoObject, NULL);
+	MonoDomain *caller_domain = mono_domain_get ();
+
+	klass = mono_class_load_from_name (image, name_space, name);
+
+	mono_handle_assign_raw (o, mono_object_new_checked (domain, klass, error));
```

Comment:
you could just move the declaration of `o` down to this line:
```c
MonoClass *klass = mono_class_load_from_name (...);
MonoObjectHandle o = MONO_HANDLE_NEW (MonoObject, mono_object_new_checked (...));
```

Reply:
Longer source line and saves approx. one instruction saved ok. Are you worried about the longer-in-time liveliness of the raw pointer? I guess it should be ok.

Reply:
That takes away the only user of mono_assign_raw then.